### PR TITLE
 Initial setup chart factory + loading data + loading profiles

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

This is the chart factory code to used when rendering charts.
The data will be passed in from gql results; waiting for the setup https://github.com/TakwimuAfrica/Takwimu.AFRICA/pull/6 .

Fixes # (issue)

## Type of change

- [x] Initial setup

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by karimkawambwa** [comment]
Screen Shot 2019-09-02 at 9 52 34 AM

Initial profile page to have more than typography

**Comment by karimkawambwa** [comment]
@KhadijaMahanga and @kilemensi kindly check PR... thanks.

**Comment by KhadijaMahanga** [comment]
> Screen Shot 2019-09-02 at 9 52 34 AM
> 
> Initial profile page to have more than typography

Is the initial page on this PR?

**Comment by karimkawambwa** [comment]
@KhadijaMahanga yes that change is in here.

**Comment by KhadijaMahanga** [comment]
@karimkawambwa please recheck this, there are no changes on `profile.js` Probably an unsuccessful `git push`

**Comment by karimkawambwa** [comment]
@KhadijaMahanga true, i pushed the changes. 

Thanks

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "eslint.enable": false
+}
```
[the remark this answers is not in the document]
Comment (karimkawambwa) [reply .vscode/settings.json]: Yeah, I have not had a chance to get back to this.

File: src/components/ProfileDetail/index.js [hunk 2]
```diff
@@ -149,205 +151,154 @@ const CountrySelector = CountrySelectorComponent;
...
+  let country;
+  if (geoLevel === 'country') {
+    const { geoCode } = geo;
+    country = config.countries.find(c => c.iso_code === geoCode);
+  } else {
+    // if level is not country, then we are in level 1
+    // const { parent_geoid: countryGeoId } = geography.this;
+    // country = config.countries.find(c => c.iso_code === countryGeoId.slice(8));
+    // country.name = geography.this.name;
   }
```
[the remark this answers is not in the document]
Comment (karimkawambwa) [reply src/components/ProfileDetail/index.js]: Lets make this on a separate PR, I want to merge this as the initial introduction to loading data. @KhadijaMahanga

File: src/Theme.js [hunk 3]
```diff
@@ -1,9 +1,59 @@
-import { createMuiTheme } from '@material-ui/core/styles';
+import { createTheme } from '@codeforafrica/hurumap-ui';
 
 const FONT_FAMILY_HEADING = '"Lora", serif';
 const FONT_FAMILY_TEXT = '"Muli", sans-serif';
 
-const Theme = createMuiTheme({
+const COLOR_BREWER_DIVERGING = [
```
Comment (kilemensi) [inline src/Theme.js]: Takwimu already has these defined here: https://github.com/TakwimuAfrica/Takwimu/blob/fdc7d603bb243553f45ab4bccc4ae66ea14cbc59/takwimu/settings.py#L206-L211

File: src/Theme.js [hunk 4]
```diff
@@ -1,9 +1,59 @@
...
+  '#a6d96a',
+  '#66bd63',
+  '#1a9850',
+  '#006837'
+].reverse();
+
+const Theme = createTheme({
+  chart: {
+    pie: {
+      legendWidth: 50,
```
Comment (kilemensi) [inline src/Theme.js]: This is kind of small

File: src/components/ChartFactory.js [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+              width={width || 400}
+              height={height}
+              data={primaryData}
+              donutLabelKey={{ dataIndex: 0, sortKey: '' }}
+            />
+          </div>
+        );
+      }
+      case 'grouped_column': {
+        console.log(primaryData);
```
Comment (kilemensi) [inline src/components/ChartFactory.js]: :eye:

File: src/data/chart.json [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+            "table": "allPopulationResidence2009S",
+            "reference": {},
+            "label": "longName",
+            "x": "populationResidence2009",
+            "y": "total"
+          }]
+        }
+      ]
+  }
+]
```
Comment (kilemensi) [inline src/data/chart.json]: New line

File: src/utils/slugify.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+export default function slugify(word) {
```
Comment (kilemensi) [inline src/utils/slugify.js]: Damn, I thought we left this bad boy behind!

File: src/pages/Profile.js [hunk 8]
```diff
@@ -1,37 +1,268 @@
...
+                <InsightContainer
+                  classes={{ root: classes.chart }}
+                  key={chart.id}
+                  loading={chartData.isLoading}
+                  title={chart.title}
+                  source={{
+                    title: 'Community Survey 2016',
+                    href: 'http://dev.dominion.africa'
+                  }}
+                >
```
Comment (kilemensi) [inline src/pages/Profile.js]: Lets add insight links for the **Read Country Analysis** button... I think it's just `/profiles/country_slug`

File: .vscode/settings.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (kilemensi) [reply .vscode/settings.json]: Still `newline` here

File: src/Theme.js (hunk 3 above)
Comment (karimkawambwa) [reply src/Theme.js]: I'll add story on PT :)

File: src/Theme.js (hunk 4 above)
Comment (karimkawambwa) [reply src/Theme.js]: Larger one makes pie charts look massive.
@KhadijaMahanga can edit this once she renders the chart definitions she is working on.

File: src/utils/slugify.js (hunk 7 above)
Comment (KhadijaMahanga) [reply src/utils/slugify.js]: behind where?

File: src/pages/Profile.js (hunk 8 above)
Comment (karimkawambwa) [reply src/pages/Profile.js]: Lets do this in a separate PR when we are sure about it.
This is just initial setup.

File: src/components/ProfileDetail/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (karimkawambwa) [reply src/components/ProfileDetail/index.js]: @KhadijaMahanga https://www.pivotaltracker.com/story/show/168316147

File: src/utils/slugify.js (hunk 7 above)
Comment (karimkawambwa) [reply src/utils/slugify.js]: @KhadijaMahanga Clemence and I had many disagreements of using slugify. But without slugify you would have to create slugs in all chart definitions.

**Pull request merged**